src/Tools/dotnet-user-jwts/src/Commands/CreateCommand.cs

```diff
@@ -0,0 +1,204 @@
+                "--expires-on",
+                @"The UTC date & time the JWT should expire in the format 'yyyy-MM-dd [[[[HH:mm]]:ss]]'. Defaults to 6 months after the --not-before date. " +
+                         "Do not use this option in conjunction with the --valid-for option.",
+                CommandOptionType.SingleValue);
+
+            var validForOption = cmd.Option(
+                "--valid-for",
+                "The period the JWT should expire after. Specify using a number followed by a period type like 'd' for days, 'h' for hours, " +
+                         "'m' for minutes, and 's' for seconds, e.g. '365d'. Do not use this option in conjunction with the --expires-on option.",
+                CommandOptionType.SingleValue);
+
+            cmd.HelpOption("-h|--help");
+
+            cmd.OnExecute(() =>
+            {
+                var (options, isValid) = ValidateArguments(
+                    app.ProjectOption, schemeNameOption, nameOption, audienceOption, issuerOption, notBeforeOption, expiresOnOption, validForOption, rolesOption, scopesOption, claimsOption);
+
+                if (!isValid)
+                {
+                    return 1;
+                }
+
+                return Execute(app.ProjectOption.Value(), options);
+            });
+        });
+    }
+
+    private static (JwtCreatorOptions, bool) ValidateArguments(
+        CommandOption projectOption,
+        CommandOption schemeNameOption,
+        CommandOption nameOption,
+        CommandOption audienceOption,
+        CommandOption issuerOption,
+        CommandOption notBeforeOption,
+        CommandOption expiresOnOption,
+        CommandOption validForOption,
+        CommandOption rolesOption,
+        CommandOption scopesOption,
+        CommandOption claimsOption)
+    {
+        var isValid = true;
+        var project = DevJwtCliHelpers.GetProject(projectOption.Value());
+        var scheme = schemeNameOption.HasValue() ? schemeNameOption.Value() : "Bearer";
+        var name = nameOption.HasValue() ? nameOption.Value() : Environment.UserName;
+
+        var audience = audienceOption.HasValue() ? audienceOption.Value() : DevJwtCliHelpers.GetApplicationUrl(project);
+        if (audience is null)
+        {
+            Console.WriteLine("Could not determine the project's HTTPS URL. Please specify an audience for the JWT using the --audience option.");
+            isValid = false;
+        }
+        var issuer = issuerOption.HasValue() ? issuerOption.Value() : DevJwtsDefaults.Issuer;
+
+        var notBefore = DateTime.UtcNow;
+        if (notBeforeOption.HasValue())
+        {
+            if (!ParseDate(notBeforeOption.Value(), out notBefore))
+            {
+                Console.WriteLine(@"The date provided for --not-before could not be parsed. Ensure you use the format 'yyyy-MM-dd [[[[HH:mm]]:ss]]'.");
+                isValid = false;
+            }
+        }
+
+        var expiresOn = notBefore.AddMonths(6);
```

Comment:
I think we landed on this being much shorter by default. I remember there being feedback that JWTs are usually only valid for minutes at a time but of course usually JWT refresh is implemented too so the refreshing is transparent to the client (usually). All that said, it's hard to know what a "good" default here is, but it feels like maybe it should be less than 6 months (28 days?).

Perhaps the tool should support setting defaults, e.g. `dotnet user-jwts defaults --valid-for 28d`, which would mean all JWTs created after that default to expire after 28 days unless a more specific value is passed (scope creep, ha!).

Reply:
Yeah, we never reached a conclusion but I'm setting this to three months for the following totally logical reasons:

- It's halfway between this current value and the smallest we've considered
- Refresh tokens expire after 90 days and although this is not a refresh token we're at least leveraging a "long enough" timespan that people who work in the space might be familiar with

Definitely something we want to inquire on via customer feedback.